tests/test_phirgen.py

```diff
@@ -10,13 +10,98 @@
 
 import json
 
-from pytket.circuit import Circuit
+from pytket.circuit import Bit, Circuit
 from pytket.phir.api import pytket_to_phir
 from pytket.qasm.qasm import circuit_from_qasm_str
+from pytket.unit_id import BitRegister
 
 from .test_utils import QasmFile, get_qasm_as_circuit
 
 
+def test_multiple_sleep() -> None:
+    """Ensure multiple sleep ops get converted correctly."""
+    qasm = """
+    OPENQASM 2.0;
+    include "hqslib1_dev.inc";
+
+    qreg q[2];
+
+    sleep(1) q[0];
+    sleep(2) q[1];
+    """
+    circ = circuit_from_qasm_str(qasm)
+    phir = json.loads(pytket_to_phir(circ))
+    assert phir["ops"][2] == {"mop": "Idle", "args": [["q", 0]], "duration": [1.0, "s"]}
+    assert phir["ops"][4] == {"mop": "Idle", "args": [["q", 1]], "duration": [2.0, "s"]}
+
+
+def test_simple_cond_classical() -> None:
+    """Ensure conditional classical operation are correctly generated."""
+    circ = get_qasm_as_circuit(QasmFile.simple_cond)
+    phir = json.loads(pytket_to_phir(circ))
+    assert phir["ops"][-6] == {"//": "IF ([c[0]] == 1) THEN SetBits(1) z[0];"}
+    assert phir["ops"][-5] == {
+        "block": "if",
+        "condition": {"cop": "==", "args": [["c", 0], 1]},
+        "true_branch": [{"cop": "=", "returns": [["z", 0]], "args": [1]}],
+    }
+
+
+def test_pytket_classical_only() -> None:
+    """From https://github.com/CQCL/pytket-phir/issues/61 ."""
+    c = Circuit(1)
+    a = c.add_c_register("a", 2)
+    b = c.add_c_register("b", 3)
+
+    c.add_c_copyreg(a, b)
+    c.add_c_copybits([Bit("b", 2), Bit("a", 1)], [Bit("a", 0), Bit("b", 0)])
+    c.add_c_copybits(
+        [Bit("b", 2), Bit("a", 1)], [Bit("a", 0), Bit("b", 0)], condition=Bit("b", 1)
+    )
+    c.add_c_copybits(
+        [Bit("a", 0), Bit("a", 1)],  # type: ignore[list-item] # overloaded function
+        [Bit("b", 0), Bit("b", 1)],  # type: ignore[list-item] # overloaded function
+        condition_bits=[Bit("b", 1), Bit("b", 2)],
+        condition_value=2,
+    )
+
+    phir = json.loads(pytket_to_phir(c))
+
+    assert phir["ops"][3] == {
+        "cop": "=",
+        "returns": [["b", 0], ["b", 1]],
+        "args": [["a", 0], ["a", 1]],
+    }
+    assert phir["ops"][5] == {
+        "cop": "=",
+        "returns": [["a", 0], ["b", 0]],
+        "args": [["b", 2], ["a", 1]],
+    }
+    assert phir["ops"][7] == {
+        "block": "if",
+        "condition": {"cop": "==", "args": [["b", 1], 1]},
+        "true_branch": [
+            {"cop": "=", "returns": [["a", 0], ["b", 0]], "args": [["b", 2], ["a", 1]]}
+        ],
+    }
+    assert phir["ops"][8] == {
+        "//": "IF ([b[1], b[2]] == 2) THEN CopyBits a[0], a[1], b[0], b[1];"
+    }
+    assert phir["ops"][9] == {
+        "block": "if",
+        "condition": {
+            "cop": "&",
+            "args": [
+                {"cop": "==", "args": [["b", 1], 1]},
+                {"cop": "==", "args": [["b", 2], 0]},
+            ],
+        },
```

Comment:
If someone can verify this is the correct translation for line 88 that'd be great.

Reply:
Ah, no it is not, because of the endianness issue mentioned above.